### PR TITLE
Trailing colons in env breaks things

### PR DESCRIPTION
**Pull request by mxcl**
Refs https://github.com/pkgxdev/pantry/pull/147

### CODE REVIEW

File: src/prefab/construct-env.ts [hunk 1]
```diff
@@ -24,6 +24,10 @@ export default async function(pkgenv: { installations: Installation[] }) {
 
   for (const key in rv) {
     rv[key] = rv[key].replaceAll(new RegExp(`\\$${key}\\b`, 'g'), `\${${key}}`)
+    // don’t end with a trailing `:` since that is sometimes interpreted as CWD and can break things
+    // instead of `foo:${PATH}` we end up with `foo${PATH:+:PATH}` which is not POSIX but works
+    // with all realy shells to avoid the trailing `:`
```
Comment (jhheider) [inline src/prefab/construct-env.ts:29]: `s/realy/really/`

File: src/prefab/construct-env.ts [hunk 2]
```diff
@@ -24,6 +24,10 @@ export default async function(pkgenv: { installations: Installation[] }) {
...
+    rv[key] = rv[key].replaceAll(new RegExp(`:\\$\{${key}}`, 'g'), `\${${key}:+:$${key}}`)
```
Comment (jhheider) [inline src/prefab/construct-env.ts:30]: this is succinct, but isn't it probably safer to do:
```javascript
  .replaceAll(/::/, ':')
  .replace(/^:/, '')
  .replace(/:$/, '')
```
?

File: src/prefab/construct-env.ts (hunk 1 above)
Comment (mxcl) [reply src/prefab/construct-env.ts:29]: lol not even sure what I meant to say. “Really” doesn't make sense

Comment (jhheider) [reply src/prefab/construct-env.ts:29]: hm, actually, it's probably `real`.

File: src/prefab/construct-env.ts (hunk 2 above)
Comment (mxcl) [reply src/prefab/construct-env.ts]: I didn't bother with some of these as we construct this env ourselves. The runtime `env` from the package.yml could in theory cause a preceding `:` I guess.

**Pull request merged**